tests/std/tests/P0627R6_unreachable/test.cpp

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cassert>
+#include <utility>
+
+constexpr int test_impl(int arg) {
+    switch (arg) {
+    case 1:
+        return 'x';
+
+    case 2:
+        return 'y';
+
+    default:
+        std::unreachable();
+    }
+}
+
+constexpr bool test() {
+    assert(test_impl(1) == 'x');
+    assert(test_impl(2) == 'y');
+    return true;
+}
+
+static_assert(test());
+
+int main() {
+    test();

```

Comment:
We conventionally put the compile-time and runtime calls of the same test function on successive lines:
```suggestion
int main() {
    test();
    static_assert(test());
```
(There maybe some old tests that aren't consistent with this, but this has been the convention for a couple of years now.)